### PR TITLE
trivial Lazy instance for functions

### PR DESCRIPTION
**Pull request by matthewleon**

### CODE REVIEW

**Comment by paf31** [comment]
👍

**Comment by matthewleon** [comment]
Might be worth waiting until I add some tests on this. Not sure it's working exactly the way I thought it would.

**Comment by matthewleon** [comment]
In particular, I'm not sure if this makes any sense given that `fix` will diverge if used directly on this instance.

Maybe it still makes sense though in those contexts in which you have a `Lazy a => Lazy (f a)`.

@paf31 perhaps you have some advise here?

**Comment by matthewleon** [comment]
Got it to stop diverging... Will update shortly.

**Comment by matthewleon** [comment]
Okay, this feels right now. Happy to have this reviewed.

File: src/Control/Lazy.purs [hunk 1]
```diff
@@ -11,7 +11,7 @@ class Lazy l where
   defer :: (Unit -> l) -> l
 
 instance lazyFn :: Lazy (a -> b) where
-  defer f = f unit
+  defer f = \x -> f unit x
```
Comment (garyb) [inline src/Control/Lazy.purs]: Now I see it, this definitely looks like the right thing 👍

**Comment by paf31** [comment]
Thanks!

**Pull request merged**